### PR TITLE
Sitemap route exclusions

### PR DESCRIPTION
**Pull request by lightstrike**
I would like to add support for excluding multiple routes from the sitemap via globbing, using `minimatch` like in #4538.

I also took a crack at addressing #4723 but it didn't work on my side. I haven't used `pathPrefix` before, maybe someone else can check?

### CODE REVIEW

File: packages/gatsby-plugin-sitemap/README.md [hunk 1]
```diff
@@ -63,4 +66,5 @@ plugins: [
 ]
 ```
 
+If you are using `pathPrefix` in `gatsby-config.js`, you must run `gatsby build --prefix-paths` to build a sitemap with the prefix included. See [path prefix docs](https://www.gatsbyjs.org/docs/path-prefix/#production-build).
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-sitemap/README.md]: Curious why was this added here? There's already a standalone doc page.

Comment (lightstrike) [reply packages/gatsby-plugin-sitemap/README.md]: It was something that tripped me up when testing, largely since I haven't used or needed `pathPrefix` before.

I need to take another shot at the `pathPrefix` stuff before this ready to merge.

Comment (KyleAMathews) [reply packages/gatsby-plugin-sitemap/README.md]: Ok, makes sense. Please remove it though as it's not relevant to this plugin other than the plugin should support it.

**Comment by lightstrike** [comment]
@KyleAMathews I took another look at the pathPrefix issue and described what I found in #4723.

I just removed the relevant `pathPrefix` code as it's not relevant to the sitemap exclusion logic and can be addressed separately.

**Comment by pieh** [comment]
I have one small request - can you add `minimatch` to `package.json`? We should declare dependency and not rely on it being installed with other packages.

**Comment by KyleAMathews** [comment]
Thanks!
